src/ImpurityDensities/BouleImpurityDensities/SplineBouleImpurityDensity.jl

```diff
@@ -16,21 +16,35 @@ struct SplineBouleImpurityDensity{N, T <: SSDFloat} <: AbstractImpurityDensity{T
     det_z0::T
 end
 
-function SplineBouleImpurityDensity{T}(z::Vector{<:Number}, ρ::Vector{<:Number}, det_z0::Number) where {T}
```

Comment:
Do you want to update going from `Vector{<:Number}` to `AbstractVector{<:RealQuantity}` also for all other `BouleImpurityDensity` constructors?

Reply:
I tested this. It doesn't work for other because there is a mix of units in the vectors